### PR TITLE
fix unnecessary Reference Check with property signature

### LINKED ISSUE #23829: Unnecessary reference checks with interfaces
cc @weswigham 

**TypeScript Version:**  master

**Search Terms:** 

**Code**

```ts
interface C {
  [C.a](): void;
  [C.b]: void;
}
class C {
  static readonly a = Symbol();
  static readonly b = Symbol();
}
```

**Expected behavior:**

pass

**Actual behavior:**

```
$ node built/local/tsc.js index.ts -t esnext
index.ts:3:4 - error TS2449: Class 'C' used before its declaration.

3   [C.b]: void;
     ~
```

**Playground Link:**  

**Related Issues:**

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #23829

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -1169,6 +1169,10 @@ namespace ts {
                         return true;
                     }
 
+                    if (isComputedPropertyName(current) && isPropertySignature(current.parent) && current.parent.name === current) {
```
Comment (mhegazy) [inline src/compiler/checker.ts]: that will also allow it inside the class body, which is not correct.

Comment (mhegazy) [reply src/compiler/checker.ts]:
```ts
class C{
    [C.a] = 0;
    static a = "a";
}
```

That would be throw based on the current TC39 proposal for class properties

**Comment by RyanCavanaugh** [comment]
Toggling to queue a fresh CI build

**Comment by RyanCavanaugh** [comment]
Red CI and unaddressed comments for several months. Feel free to open a fresh PR as usual @Kingwl 👍

**Pull request closed without merging**